### ISSUE TITLE
questions about test and tutorial results

### ISSUE THREAD

**Issue by danielskatz** [AlexPof/opycleid#2]
When I run the test in the README and the functions in the tutorial, I get slightly different answers in some cases - see the notebook at http://danielskatz.org/testing_opycleid.ipynb, which has what I ran and my results.

For example, 
Out[2] = ['T9', 'I9']
but the README says the output should be 
['I9', 'T9']

why are these backwards?

Similarly, Out[14] and Out[15]... and Out[16] and maybe Out[17]

**Comment by AlexPof** [comment]
Thank you for pointing out this behavior. I believe this is due to the platform-dependent implementation of *dict.items()* in Python (I developed Opycleid and wrote the corresponding tutorial on Os X, and you might have executed on another OS). If the keys are enumerated in a different order, the morphisms will end up with different names, although the transformations they represent are nevertheless the same.

In order to standardize this and to avoid discrepancies between OS, I have made some changes to the code, in which the keys are now sorted during the enumeration. I have rerun the tests and tutorials on OS X, and I have modified the ReadMe and the docs with the new results it gave.

If you please reinstall Opycleid and run the tests and tutorials again, there should be no more differences. Please note that you will have to reinstall Opycleid manually from the GitHub repository, as I have not yet reuploaded a new version on PyPi since there might still be additional changes to be made.

**Comment by danielskatz** [comment]
I was actually using OS X too.  But I'll try again and see what happens.

**Comment by danielskatz** [comment]
Perhaps you can add non PyPi build instructions to the README and docs?

**Comment by danielskatz** [comment]
Specifically, I did
```
git clone ...
cd opycleid
python setup.py build
sudo python setup.py install
```

The install fails for me with
```
File "/private/tmp/easy_install-3rbAOv/numpy-1.15.4/numpy/core/setup.py", line 14, in <module>

ImportError: No module named _build_utils.apple_accelerate
```

Some simple numpy testing makes it clear than my numpy is ok.

**Comment by danielskatz** [comment]
Ok, I rebuilt.  This clears up the differences in my cells Out[2], Out[16], Out[17] but not Out[14] and Out[15], where I see

```from opycleid.categoryaction import CategoryAction

my_category = CategoryAction()
my_category.set_objects([X,Y])
my_category.add_generators([T,I,J])
my_category.generate_category()
for x in my_category.operations:
  print(x)
```

```
T
同
和
id_X
id_Y
TT
同T
和同
T和
TTT
同TT
TT和
同T和
和同T
TTTT
同TTT
TTT和
同TT和
和同TT
TTTTT
同TTT和
和同TTT
TTTTTT
TTTTTTT
TTTTTTTT
TTTTTTTTT
TTTTTTTTTT
TTTTTTTTTTT
```

and

```
my_category.rewrite_operations()
for x in my_category.operations:
  print(x)
```

```
T
同
和
id_X
id_Y
(T^2)
同T
和同
T和
(T^3)
同(T^2)
(T^2)和
同T和
和同T
(T^4)
同(T^3)
(T^3)和
同(T^2)和
和同(T^2)
(T^5)
同(T^3)和
和同(T^3)
(T^6)
(T^7)
(T^8)
(T^9)
(T^10)
(T^11)
```

which does not match what the [tutorial](https://alexpof.github.io/opycleid/gettingstarted/#defining-custom-morphisms-and-category-actions) says I should get.

**Comment by AlexPof** [comment]
This is my bad, I didn't realize the same problem I was fixing was also present in the tutorial (in ```for x in my_category.operations```).

To avoid accessing directly the internal dictionaries, I have added new methods, among which ```get_morphisms()```, and corrected the tutorial accordingly. The documentation should also be up to date regarding these new methods. Do not hesitate to tell me if the tests match the tutorial.

As before, I have committed the changes in the repository, but did not upload a new version to PyPi.
